### PR TITLE
migrate to github actions

### PR DESCRIPTION
**Pull request by ktoso**

### CODE REVIEW

File: .github/workflows/pull_request.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+jobs:
+  soundness:
+    name: Soundness
+    uses: swiftlang/github-workflows/.github/workflows/soundness.yml@main
+    with:
+      license_header_check_project_name: "RediStack"
+
+  unit-tests:
+    name: Unit tests
+    uses: apple/swift-nio/.github/workflows/unit_tests.yml@main
```
Comment (0xTim) [inline .github/workflows/pull_request.yml:16]: Do we want to use the swift lang org workflow instead?

Comment (ktoso) [reply .github/workflows/pull_request.yml:16]: Yeah that's a good point

Comment (ktoso) [reply .github/workflows/pull_request.yml:16]: Hm looking at various repos we have a mix of which ones we use... I may stick with this for now, we'll see.

**Comment by ktoso** [comment]
Not entirely done here but need to leave for now, I'll continue later.

**Pull request merged**